homeassistant/components/plaato/__init__.py

```diff
@@ -96,35 +223,42 @@ async def handle_webhook(hass, webhook_id, request):
         return
 
     device_id = _device_id(data)
+    sensor_data = PlaatoAirlock.from_web_hook(data)
 
-    attrs = {
-        ATTR_DEVICE_NAME: data.get(ATTR_DEVICE_NAME),
-        ATTR_DEVICE_ID: data.get(ATTR_DEVICE_ID),
-        ATTR_TEMP_UNIT: data.get(ATTR_TEMP_UNIT),
-        ATTR_VOLUME_UNIT: data.get(ATTR_VOLUME_UNIT),
-    }
-
-    sensors = {
-        ATTR_TEMP: data.get(ATTR_TEMP),
-        ATTR_BPM: data.get(ATTR_BPM),
-        ATTR_SG: data.get(ATTR_SG),
-        ATTR_OG: data.get(ATTR_OG),
-        ATTR_ABV: data.get(ATTR_ABV),
-        ATTR_CO2_VOLUME: data.get(ATTR_CO2_VOLUME),
-        ATTR_BATCH_VOLUME: data.get(ATTR_BATCH_VOLUME),
-        ATTR_BUBBLES: data.get(ATTR_BUBBLES),
-    }
-
-    hass.data[DOMAIN][device_id] = {
-        PLAATO_DEVICE_ATTRS: attrs,
-        PLAATO_DEVICE_SENSORS: sensors,
-    }
-
-    async_dispatcher_send(hass, SENSOR_UPDATE, device_id)
+    async_dispatcher_send(hass, SENSOR_UPDATE, *(device_id, sensor_data))
 
     return web.Response(text=f"Saving status for {device_id}", status=HTTP_OK)
 
 
 def _device_id(data):
     """Return name of device sensor."""
     return f"{data.get(ATTR_DEVICE_NAME)}_{data.get(ATTR_DEVICE_ID)}"
+
+
+class PlaatoCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from the API."""
+
+    def __init__(self, hass, auth_token, device_type: PlaatoDeviceType):
+        """Initialize."""
+        self.api = Plaato(auth_token=auth_token)
+        self.hass = hass
+        self.device_type = device_type
+        self.platforms = []
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+
+    async def _async_update_data(self):
+        """Update data via library."""
+        try:
+            data = await self.api.get_data(
+                session=aiohttp_client.async_get_clientsession(self.hass),
+                device_type=self.device_type,
+            )
+            return data
+        except Exception as exception:
```

Comment:
You can remove the try as DataUpdateCoordinator will already catch the exception.